fideslog/sdk/python/client.py

```diff
@@ -91,17 +73,42 @@ def register(self, registration: Registration) -> None:
         Register a new user.
         """
 
-        __set_event_loop()
+        self.__set_event_loop()
         run(self.__send(registration))
 
     def send(self, event: AnalyticsEvent) -> None:
         """
         Record a new analytics event.
         """
 
-        __set_event_loop()
+        self.__set_event_loop()
         run(self.__send(event))
 
+    def __set_event_loop(self) -> None:
+        """
+        Helps to work around a bug in the default Windows event loop for Python 3.8+
+        by changing the default event loop in Windows processes.
+        """
+
+        if self.event_loop_set:
+            # Exit early if this has already run to avoid setting the event loop
+            # multiple times.
+            return
+
+        if (
+            version_info[0] == 3
+            and version_info[1] >= 8
+            and platform.lower().startswith("win")
+        ):
+            from asyncio import (  # type: ignore[attr-defined]
+                WindowsSelectorEventLoopPolicy,
+                set_event_loop_policy,
+            )
+
+            set_event_loop_policy(WindowsSelectorEventLoopPolicy())
+
+        self.event_loop_set = True
```

Comment:
This is set deliberately outside the `if` block above so that we don't indulge checks on systems that don't require the event loop to be set.